chain-vote/src/cryptography/zkps/correct_decryption/correct_decryption_zkp.rs

```diff
@@ -0,0 +1,116 @@
+//! Non-interactive Zero Knowledge proof for correct ElGamal
```

Comment:
The Treasury voting spec is going to be modified so that the proof of ElGamal decryption will be the same as the DLEQ proof. I would keep the latest docs, and (unfortunately) repeat the DLEQ proof code here.